cosmwasm/enclaves/shared/contract-engine/src/contract_operations.rs

```diff
@@ -223,6 +259,300 @@ fn to_canonical(contract_address: &BaseAddr) -> Result<CanonicalAddr, EnclaveErr
     })
 }
 
+lazy_static::lazy_static! {
+    static ref HARDCODED_CONTRACT_ADMINS: HashMap<&'static str, &'static str> = HashMap::from([
+        (
+            "secret1exampleContractAddress1",
+            "secret1ExampleAdminAddress1",
+        ),
+        (
+            "secret1exampleContractAddress2",
+            "secret1ExampleAdminAddress2",
+        ),
+    ]);
+}
+
+fn is_hardcoded_contract_admin(
+    contract: &CanonicalAddr,
+    admin: &CanonicalAddr,
+    admin_proof: &[u8],
+) -> bool {
+    if admin_proof != [0; enclave_crypto::HASH_SIZE] {
+        return false;
+    }
+
+    let contract = HumanAddr::from_canonical(contract);
+    if contract.is_err() {
+        trace!(
+            "is_hardcoded_contract_admin: failed to convert contract to human address: {:?}",
+            contract.err().unwrap()
+        );
+        return false;
+    }
+    let contract = contract.unwrap();
+
+    let admin = HumanAddr::from_canonical(admin);
+    if admin.is_err() {
+        trace!(
+            "is_hardcoded_contract_admin: failed to convert admin to human address: {:?}",
+            admin.err().unwrap()
+        );
+        return false;
+    }
+    let admin = admin.unwrap();
+
+    HARDCODED_CONTRACT_ADMINS.get(contract.as_str()) == Some(&admin.as_str())
+}
+
+#[cfg_attr(feature = "cargo-clippy", allow(clippy::too_many_arguments))]
+pub fn migrate(
+    context: Ctx,
+    gas_limit: u64,
+    used_gas: &mut u64,
+    contract: &[u8],
+    env: &[u8],
+    msg: &[u8],
+    sig_info: &[u8],
+    admin: &[u8],
+    admin_proof: &[u8],
+) -> Result<MigrateSuccess, EnclaveError> {
+    debug!("Starting migrate");
+
+    //let start = Instant::now();
+    let contract_code = ContractCode::new(contract);
+    let contract_hash = contract_code.hash();
+    // let duration = start.elapsed();
+    // trace!("Time elapsed in ContractCode::new is: {:?}", duration);
+    debug!(
+        "******************** migrate RUNNING WITH CODE: {:x?}",
+        contract_hash
+    );
+
+    //let start = Instant::now();
+    let base_env: BaseEnv = extract_base_env(env)?;
+
+    #[cfg(feature = "light-client-validation")]
+    {
+        verify_block_info(&base_env)?;
+    }
+
+    // let duration = start.elapsed();
+    // trace!("Time elapsed in extract_base_env is: {:?}", duration);
+    let query_depth = extract_query_depth(env)?;
+
+    //let start = Instant::now();
+    let (sender, contract_address, block_height, sent_funds) = base_env.get_verification_params();
+    // let duration = start.elapsed();
+    // trace!("Time elapsed in get_verification_paramsis: {:?}", duration);
+
+    let canonical_contract_address = to_canonical(contract_address)?;
+    let canonical_sender_address = to_canonical(sender)?;
+    let canonical_admin_address = CanonicalAddr::from_vec(admin.to_vec());
+
+    let og_contract_key = base_env.get_og_contract_key()?;
```

Comment:
Yes, the admin proof will pass and you'll be able to migrate offline to a malicious code. Unfortunately you'll not be able to access the state with that code because you need the correct `og_contract_key`, but if you'll pass the correct one in a subsequent call it'll fail on [this check](https://github.com/scrtlabs/SecretNetwork/blob/280ed5f62/cosmwasm/enclaves/shared/contract-engine/src/contract_validation.rs#L262-L272) when it tries to validate current_contract_key against og_contract_key.